modules/codegen/src/main/scala/com/twilio/guardrail/generators/Http4sServerGenerator.scala

```diff
@@ -501,8 +502,14 @@ object Http4sServerGenerator {
           case Nil        => responseInMatch
           case generators => q"for {..${generators :+ enumerator"response <- $responseInMatch"}} yield response"
         }
-        val routeBody       = entityProcessor.fold[Term](responseInMatchInFor)(_.apply(responseInMatchInFor))
-        val fullRoute: Case = p"case req @ $fullRouteWithTracingMatcher => $routeBody"
+        val routeBody = entityProcessor.fold[Term](responseInMatchInFor)(_.apply(responseInMatchInFor))
+        val fullRoute: Case =
+          p"""case req @ $fullRouteWithTracingMatcher => 
+             val response = {
```

Comment:
This is really good -- The one consideration is that we are trying to reduce top-level bindings inside of an individual route. The way things are written currently, parameters named `response` will now conflict.

Reply:
good point, i can skip the `val response` and just add the computation into the tuple directly, i did it this way to make the code nicer but it's not required to be this way